doc/specs/Keyboard-Selection.md

```diff
@@ -63,6 +68,39 @@ Every combination of the `Direction` and `SelectionExpansionMode` will map to a
 **NOTE**: If `copyOnSelect` is enabled, we need to make sure we update the clipboard on every change in selection.
 
 
+
+### Mark Mode
+Mark Mode is a mode where the user can begin a selection using only the keyboard. It can be cycled into an enabled/disabled state through a user keybinding (`toggleMarkMode`).
+
+Upon being enabled, the current cursor position becomes the selection endpoint. The user can move the selection anchor using the same keybindings defined for keyboard selection (`moveSelectionPoint` action).
+
+#### `toggleMarkMode`
+`toggleMarkMode` has two keybinding arguments:
+| Parameter | Accepted Values | Default Value | Description |
+|--|--|--|--|
+| `anchorModifier` | regular keybindings and plain modifier keys | `Shift` | The keybinding used to anchor the selection endpoint to it's current text buffer position. While holding down that keybinding, any `moveSelectionPoint` actions will move the other selection anchor (`endSelectionPosition`). |
+| `anchorMode` | `Hold`, `Toggle` | `Hold` | If `Hold`, the user needs to hold the `anchorModifier` key to anchor the 'start' selection endpoint and move the second selection endpoint (like Mark Mode). If `Toggle`, the user switches between selection endpoints when they press it (like Copy Mode). |
+
+#### Rendering during Mark Mode
+Since we are just moving the selection anchors, rendering the selection rects should operate normally. We need to ensure that we still scroll when we move a selection anchor past the top/bottom of the viewport.
+
+In ConHost, output would be paused when a selection was present. This is a completely separate issue that is being tracked in (#2529)[https://github.com/microsoft/terminal/pull/2529].
+
+#### Interaction with CopyOnSelect
+If `copyOnSelect` is enabled, the selection is copied when the selection operation is "complete". If `anchorMode=Hold`, the user has to use the `copy` keybinding to signify that they have finished creating a selection. If `anchorMode=Toggle`, the selection is copied either when the `copy` keybinding is used, or when the user presses the `anchorModifier` key and the 'end' endpoint is set.

```

Comment:
I might be misunderstanding the last part of this section, but it sounds like it'll only copy if the user hits the `anchorModifier` key and that one keypress anchors the `end` endpoint, but not if it anchors the `start` endpoint. It should probably copy no matter which endpoint is set right?

Reply:
So, my concern here is that we shouldn't be copying too often. With mouse selection, it's easy because we know a selection is "complete" when the user releases the left button. Abstracting that logic to Mark Mode (in anchorMode=toggle),...
- enter mark mode --> X (you just use the mouse)
- switch selection anchor target --> press/hold left mouse button
- move 'end' anchor --> drag mouse
- switch selection anchor target --> release left mouse button